core/corehttp/routing.go

```diff
@@ -96,6 +96,57 @@ func (r *contentRouter) PutIPNS(ctx context.Context, name ipns.Name, record *ipn
 	return r.n.Routing.PutValue(ctx, string(name.RoutingKey()), raw)
 }
 
+func (r *contentRouter) GetClosestPeers(ctx context.Context, pid peer.ID) (iter.ResultIter[*types.PeerRecord], error) {
+	// Per the spec, if the peer ID is empty, we should use self.
+	if pid == "" {
+		pid = r.n.Identity
+	}
+
+	peers, err := r.n.DHT.WAN.GetClosestPeers(ctx, string(pid))
+	if err != nil {
+		return nil, err
+	}
+
+	lanPeers, err := r.n.DHT.LAN.GetClosestPeers(ctx, string(pid))
+	if err != nil {
+		return nil, err
+	}
+	peers = append(peers, lanPeers...)
+
+	// We have some DHT-closest peers. Find addresses for them.  We can
+	// use any routers for that, we can find records for DHT peers on
+	// non-DHT routers with whatever protocols. FIXME: right? right??
+	var records []*types.PeerRecord
+	for _, p := range peers {
+		record := types.PeerRecord{
+			ID:     &p,
+			Schema: types.SchemaPeer,
+			// we dont seem to care about protocol/extra infos
+			// FIXME: should FindPeers care? That info seems to
+			// not cross the FindPeer API.
+		}
```

Comment:
After a call to `GetClosestPeers()`, the peers addresses can be found in the peerstore (`r.n.Peerstore`), both for the accelerated DHT client and normal client (used in dual). No need to make new network calls.